stdlib/runtime/equal.gr

```diff
@@ -0,0 +1,209 @@
+/* compilation-mode: runtime */
```

Comment:
Mostly taking your word for this conversion. I don't have the energy to try to line up the AssemblyScript side-by-side with the Grain code.